speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DataCompressionMenu.cpp

```diff
@@ -0,0 +1,233 @@
+static void LoadSettingsFromFile(void* p_param)
+{
+    m_dataCompressionType = std::stoi(GfParmGetStr(p_param, PRM_DATA_COMPRESSION, GFMNU_ATTR_SELECTED, "1"));
+    m_customCompressionLevel = std::stoi(GfParmGetStr(p_param, PRM_CUSTOM_COMPRESSION, GFMNU_ATTR_TEXT, nullptr));
+}
+
+/// @brief Makes sure all visuals display the internal values
+static void SynchronizeControls()
+{
+    GfuiRadioButtonListSetSelected(s_scrHandle, m_compressionButtonList, static_cast<int>(m_dataCompressionType));
+
+    char buf[32];
+    sprintf(buf, "%d", m_customCompressionLevel);
+    GfuiEditboxSetString(s_scrHandle, m_customCompressionLevelControl, buf);
+}
+
+/// @brief Loads default settings
+static void LoadDefaultSettings()
+{
+    m_dataCompressionType = GfuiRadioButtonListGetSelected(s_scrHandle, m_compressionButtonList);
+    m_customCompressionLevel = std::stoi(GfuiEditboxGetString(s_scrHandle, m_customCompressionLevelControl));
+}
+
+/// @brief Loads (if possible) the settings; otherwise, the control's default settings will be used
+static void LoadSettings()
+{
+    char buf[512];
+    sprintf(buf, "%s%s", GfLocalDir(), COMPRESSION_FILEPATH);
+    if (GfFileExists(buf))
+    {
+        void* param = GfParmReadFile(buf, GFPARM_RMODE_STD);
+        // Initialize settings with the retrieved xml file
+        LoadSettingsFromFile(param);
+        SynchronizeControls();
+        return;
+    }
+    LoadDefaultSettings();
+}
+
+/// @brief Saves the settings to a file
+static void SaveSettingsToFile()
+{
+    std::string dstStr(COMPRESSION_FILEPATH);
+    char dst[512];
+    sprintf(dst, "%s%s", GfLocalDir(), dstStr.c_str());
+    void* readParam = GfParmReadFile(dst, GFPARM_RMODE_REREAD | GFPARM_RMODE_CREAT);
+
+    // Write data compression type
+    char val[32];
+    sprintf(val, "%d", m_dataCompressionType);
+    GfParmSetStr(readParam, PRM_DATA_COMPRESSION, GFMNU_ATTR_SELECTED, val);
+
+    // Save custom compression to xml file
+    char buf[32];
+    sprintf(buf, "%d", m_customCompressionLevel);
+    GfParmSetStr(readParam, PRM_CUSTOM_COMPRESSION, GFMNU_ATTR_TEXT, buf);
+
+    // Write queued changes
+    GfParmWriteFile(nullptr, readParam, COMPRESSION_SCREEN_NAME);
+}
+
+/// @brief Saves the settings so the mediator (or future instances) can access them
+static void SaveSettings()
+{
+    int compressionLevel;
+    switch (m_dataCompressionType)
+    {
+        case COMPRESSION_NONE:
+        {
+            compressionLevel = 1;
+            break;
+        }
+        case COMPRESSION_MINIMUM:
+        {
+            compressionLevel = 3;
+            break;
+        }
+        case COMPRESSION_MEDIUM:
+        {
+            compressionLevel = 5;
+            break;
+        }
+        case COMPRESSION_MAXIMUM:
+        {
+            compressionLevel = 9;
+            break;
+        }
+        default:  // Custom
+        {
+            compressionLevel = m_customCompressionLevel;
+            break;
+        }
```

Comment:
Should be `case: COMPRESSION_CUSTOM:` instead of a catch-all in the default case. In this switch you know all the cases that are valid, so if it's any other value it should (log an error: `GfLogError("some message")` and set compression to some default value, or just simply throw an error)